### PR TITLE
Cache views plugin data

### PR DESCRIPTION
**Pull request by klonos**
https://www.drupal.org/node/1376400

https://git.drupalcode.org/project/views/commit/580ff35b315b8cf0eed6e33ab97e985634f829b1

https://github.com/backdrop/backdrop-issues/issues/3719

### CODE REVIEW

File: core/modules/views/includes/cache.inc [hunk 1]
```diff
@@ -127,8 +127,25 @@ function _views_data_process_entity_types(&$data) {
...
-    $cache = views_discover_plugins();
+    // Load necessary code once.
+    if (!isset($cache)) {
+      views_include('plugins');
+      views_include_handlers();
+    }
+    // Because plugin data contains translated strings, and as such can be
+    // expensive to build, the results are cached per language.
+    global $language;
+    $cache_key = 'views:plugin_data:' . $language->langcode;
```
Comment (klonos) [inline core/modules/views/includes/cache.inc:138]: Wondering if we should be checking if `$language->langcode` is set here 🤔

Comment (quicksketch) [reply core/modules/views/includes/cache.inc:138]: It is always set  on a full bootstrap. I think this is safe.

**Comment by quicksketch** [comment]
Not sure why this one is failing tests. I restarted it to check.

File: core/modules/views/includes/cache.inc [hunk 2]
```diff
@@ -127,8 +127,25 @@ function _views_data_process_entity_types(&$data) {
 function _views_fetch_plugin_data($type = NULL, $plugin = NULL, $reset = FALSE) {
   static $cache = NULL;
   if (!isset($cache) || $reset) {
-    views_include_handlers();
-    $cache = views_discover_plugins();
+    // Load necessary code once.
+    if (!isset($cache)) {
+      views_include('plugins');
```
Comment (quicksketch) [inline core/modules/views/includes/cache.inc:132]: I think this line is unnecessary and might be the reason for the failure?

```suggestion
```

**Comment by quicksketch** [comment]
Gah, I accidentally merged this one even though it's failing tests. I've reverted it (https://github.com/backdrop/backdrop/pull/2801) and I'll merge https://github.com/backdrop/backdrop/pull/2800 instead.

**Pull request merged**